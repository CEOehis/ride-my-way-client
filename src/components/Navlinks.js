import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Navlinks = ({ user, isAuthenticated, signOut }) => {
  if (isAuthenticated) {
    return (
      <Fragment>
        <li><Link className="brand" to="/">RideMyWay</Link></li>
        <li><Link to="/create">Offer a Ride</Link></li>
        <li><Link to="/home">View Rides</Link></li>
        <li className="dropdown">
          <a>
            {
              user.fullName
                ? `Hi, ${user.fullName.split(' ')[0]}`
                : <i className="fa fa-spinner fa-spin fa-pulse" />
            }
            <i className="fa fa-chevron-down" />
          </a>
          <ul className="dropdown-content">
            <li><Link to="/profile">Profile</Link></li>
            <li><Link to="/user/rides">My Ride Offers</Link></li>
            <li><a onClick={signOut}>Sign out</a></li>
          </ul>
        </li>
      </Fragment>
    );
  }

  return (
    <Fragment>
      <li>
        <Link to="/" className="brand">RideMyWay</Link>
      </li>
      <li>
        <Link to="/register">Register</Link>
      </li>
      <li>
        <Link to="/login">Login</Link>
      </li>
    </Fragment>
  );
};

Navlinks.propTypes = {
  user: PropTypes.shape({
    email: PropTypes.string,
    fullName: PropTypes.string,
    phone: PropTypes.string,
    userId: PropTypes.number,
  }),
  signOut: PropTypes.func,
  isAuthenticated: PropTypes.bool.isRequired,
};

Navlinks.defaultProps = {
  signOut: () => null,
};

Navlinks.defaultProps = {
  user: undefined,
};

export default Navlinks;
