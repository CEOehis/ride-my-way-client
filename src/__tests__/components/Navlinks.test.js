import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navlinks from '../../components/Navlinks';

describe('Form submit Navlinks component', () => {
  it('should display links for logged in user', () => {
    const wrapper = shallow(<Navlinks user={{ fullName: 'user' }} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should display links login/register links', () => {
    const wrapper = shallow(<Navlinks />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
