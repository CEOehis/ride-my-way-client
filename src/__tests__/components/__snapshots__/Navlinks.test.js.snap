// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form submit Navlinks component should display links for logged in user 1`] = `
<Fragment>
  <li>
    <Link
      className="brand"
      replace={false}
      to="/"
    >
      RideMyWay
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/create"
    >
      Offer a Ride
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/home"
    >
      View Rides
    </Link>
  </li>
  <li
    className="dropdown"
  >
    <a>
      Hi,
       user
      <i
        className="fa fa-chevron-down"
      />
    </a>
    <ul
      className="dropdown-content"
    >
      <li>
        <Link
          replace={false}
          to="/profile"
        >
          Profile
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/user/rides"
        >
          My Ride Offers
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/logout"
        >
          Sign out
        </Link>
      </li>
    </ul>
  </li>
</Fragment>
`;

exports[`Form submit Navlinks component should display links login/register links 1`] = `
<Fragment>
  <li>
    <Link
      className="brand"
      replace={false}
      to="/"
    >
      RideMyWay
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/register"
    >
      Register
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </li>
</Fragment>
`;
